spec/integration/indexer/indexing_spec.rb

```diff
@@ -125,4 +125,41 @@
     expect(last_response.status).to eq(202)
     expect_document_is_in_rummager(SAMPLE_DOCUMENT)
   end
+
+  it "rescheduling during locking does not corrupt the process" do
+    stubbed_client = client
+
+    locked_response = { "items" => [
+      { "index" => { "error" => { "reason" => "[FORBIDDEN/metasearch/index write" } } }
+    ] }
+
+    expect(stubbed_client).to receive(:bulk).and_return(locked_response)
+    expect(stubbed_client).to receive(:bulk).and_call_original
+    allow_any_instance_of(SearchIndices::Index).to receive(:build_client).and_return(stubbed_client)
+
+    details = <<~DETAILS
+      {\"best_bets\":[
+        {\"link\":\"/learn-to-drive-a-car\",\"position\":1},
+        {\"link\":\"/learn-to-drive-a-car\",\"position\":3},
+        {\"link\":\"/learn-to-drive-a-car\",\"position\":10}
+      ],\"worst_bets\":[]}", "stemmed_query_as_term"=>" learn to drive "}]
+    DETAILS
+
+    post "/metasearch_test/documents", {
```

Comment:
I wonder if it would be worth adding a similar test for the content indexing endpoint? I know the process is (currently) the same under the hood, but it would be good to have more confidence that updating a document isn't affected by locking, because that's even more important than best bets.

Reply:
its the same piece of code so that "should" be covered by this test..

As a side note we did corrupt some data on friday due to index locking.. :(

Reply:
Yeah, I know it's the same piece of code, and you shouldn't (and can't) test lots of combinations of things in integration tests. But the implementation can change, and I'm just wondering if it's worth adding a second test here because it's really important that the _content_ indices aren't affected by locking, not just the metasearch index?

Is that data corrupted or just stale? Can we fix it with the Gor traffic replay?

Reply:
A test for content has been added. This has shown that the issue only occurs for bets-bet updates as everything else already creates a copy of the hash before the `_id` and `_type` delete